2-Draft-Accepted/RFCxxxx-PSCredential-Improvements.md

```diff
@@ -0,0 +1,102 @@
+---
+RFC: RFCxxxx
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: Security
+Comments Due: June 30th, 2019
+Plan to implement: Yes, PS7
+---
+
+# Simplify and secure use of credentials in scripts
+
+Advanced scripts that touch many systems require multiple credentials and
+types of credentials particularly when orchestrating across different clouds.
+Best practice is to not hard code any credentials in scripts, but currently
+this requires custom code on different platforms to handle this securely.
+
+This RFC proposes new cmdlets to make this simpler and secure.
+
+## Motivation
+
+> As a PowerShell script author,
+> I can securely use multiple credentials,
+> so that I can automate complex orchestration across multiple remote resources.
+
+## User Experience
+
+Registering credentials for later use:
+
+```powershell
+Install-Module AzVault
```

Comment:
Yes, Az.KeyVault, will update the spec to match that module

Reply:
Actually, the current Az.KeyVault module doesn't fit into the extension model I have in mind, it requires Connect-AzAccount to authenticate instead of having a `-Credential` or `-AccessToken` parameter to authenticate.  We will need to work with AzPS team to add that capability directly into that cmdlet.